src/ert/_c_wrappers/enkf/config/gen_kw_config.py

```diff
@@ -200,3 +162,178 @@ def sample_values(
                 values = values[active_realizations]
             parameter_values.append(values)
         return np.array(parameter_values)
+
+    @staticmethod
+    def parse_transfer_function(param_string: str) -> TransferFunction:
+        param_args = param_string.split()
+
+        TRANS_FUNC_ARGS: Final = {
+            "NORMAL": ["MEAN", "STD"],
+            "LOGNORMAL": ["MEAN", "STD"],
+            "TRUNCATED_NORMAL": ["MEAN", "STD", "MIN", "MAX"],
+            "TRIANGULAR": ["XMIN", "XMODE", "XMAX"],
+            "UNIFORM": ["MIN", "MAX"],
+            "DUNIF": ["STEPS", "MIN", "MAX"],
+            "ERRF": ["MIN", "MAX", "SKEWNESS", "WIDTH"],
+            "DERRF": ["STEPS", "MIN", "MAX", "SKEWNESS", "WIDTH"],
+            "LOGUNIF": ["MIN", "MAX"],
+            "CONST": ["VALUE"],
+            "RAW": [],
+        }
+
+        if len(param_args) > 1:
+            func_name = param_args[0]
+            param_func_name = param_args[1]
+
+            if (
+                param_func_name not in TRANS_FUNC_ARGS
+                or param_func_name not in PRIOR_FUNCTIONS
+            ):
+                raise ConfigValidationError(
+                    f"Unknown transfer function provided: {param_func_name}"
+                )
+
+            param_names = TRANS_FUNC_ARGS[param_func_name]
+
+            if len(param_args) - 2 != len(param_names):
+                raise ConfigValidationError(
+                    f"Incorrect number of values provided: {param_string}"
+                )
+
+            param_floats = []
+            try:
+                for p in param_args[2:]:
+                    param_floats.append(float(p))
+            except ValueError:
+                raise ConfigValidationError(f"Unable to convert float number: {p}")
```

Comment:
Not sure if it is a problem, but if `p` never gets a value in the `try..`   - would the error still make sense?

Reply:
Good catch. This should be fixed. :+1: